packages/client/src/connection.ts

```diff
@@ -35,6 +44,13 @@ export interface ConnectionOptions {
    */
   tls?: TLSConfig | boolean | null;
 
+  /**
+   * Proxying configuration.
+   *
+   * @experimental
+   */
+  proxy?: ProxyConfig;
```

Comment:
My concern here is it's unclear that this is HTTP CONNECT proxy (as opposed to SOCKS or something). That's why in Core and Python I was explicit w/ the field name that it's for HTTP CONNECT proxy. Granted other places just assume this is what is meant by "proxy" but I think it's better if we make it clear.

Reply:
Hence, `ProxyConfig` has a `type` property, with value `http-connect`. If we are to add support for other proxy protocols at some point (e.g. socks5), I'd prefer not to pollute the connection options with a different property for each of them.

Reply:
I see. Arguably a bit rough to have to provide a string literal in addition to the field knowing we probably will never support another type. Basically we traded willingness for our connection options to have multiple fields for a user to have to provide multiple fields, but no biggie. So long as clear the to user, works for me.